src-cpp/backend/wallet_accounts.cpp

```diff
@@ -13,9 +13,63 @@ RpcResponse<QJsonArray> getAccounts()
 {
     QJsonObject inputJSON{{"jsonrpc", "2.0"}, {"method", "accounts_getAccounts"}, {"params", QJsonValue()}};
     auto result = CallPrivateRPC(Utils::jsonToStr(inputJSON).toUtf8().data());
-    return RpcResponse<QJsonArray>(result, QJsonDocument::fromJson(result)["result"].toArray());
+    auto obj = QJsonDocument::fromJson(result).object();
+    Backend::Utils::throwOnError(obj);
+    return RpcResponse<QJsonArray>(result, obj["result"].toArray(), RpcError(-1, QJsonDocument::fromJson(result)["error"].toString()));
 }
 
+RpcResponse<QString> generateNewAccount(QString password, QString accountName, QString color)
+{
+    QString hashedPassword(Backend::Utils::hashString(password));
+    QJsonArray payload = {hashedPassword, accountName, color};
+    QJsonObject inputJSON{{"jsonrpc", "2.0"}, {"method", "accounts_generateAccount"}, {"params", payload}};
+    auto result = CallPrivateRPC(Utils::jsonToStr(inputJSON).toUtf8().data());
+    auto response = QJsonDocument::fromJson(result);
+    return  RpcResponse<QString>(result,response["result"].toString(),
+            RpcError(response["error"]["code"].toDouble(), response["error"]["message"].toString()));
```

Comment:
what if those properties not exists?

Reply:
so double conversion  - from `double` to `int` and from `int` again to `double`. Are you sure error code type is `double`?

Reply:
I printed it and found it to be like -323121212 something like this

Reply:
> what if those properties not exists?

so the thing is in this specific case it does exist, this sadly we need to depend on the status_go response and it is filled differently for different apis.`
And for each of them we need to add it this way.